timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosLeadershipCreator.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.timelock.paxos;
+
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import com.google.common.base.Preconditions;
+import com.palantir.atlasdb.config.ImmutableLeaderConfig;
+import com.palantir.atlasdb.config.LeaderConfig;
+import com.palantir.atlasdb.factory.ImmutableRemotePaxosServerSpec;
+import com.palantir.atlasdb.factory.Leaders;
+import com.palantir.atlasdb.timelock.paxos.LeadershipResource;
+import com.palantir.atlasdb.timelock.paxos.PaxosTimeLockConstants;
+import com.palantir.atlasdb.timelock.paxos.PaxosTimeLockUriUtils;
+import com.palantir.leader.LeaderElectionService;
+import com.palantir.leader.proxy.AwaitingLeadershipProxy;
+import com.palantir.timelock.Observables;
+import com.palantir.timelock.config.ImmutablePaxosRuntimeConfiguration;
+import com.palantir.timelock.config.PaxosInstallConfiguration;
+import com.palantir.timelock.config.PaxosRuntimeConfiguration;
+import com.palantir.timelock.config.TimeLockInstallConfiguration;
+import com.palantir.timelock.config.TimeLockRuntimeConfiguration;
+
+import io.reactivex.Observable;
+
+public class PaxosLeadershipCreator {
+    private final TimeLockInstallConfiguration install;
+    private final PaxosInstallConfiguration paxosInstall;
+    private final Observable<Optional<PaxosRuntimeConfiguration>> runtime;
+    private final Consumer<Object> registrar;
+
+    private LeaderElectionService leaderElectionService;
+
+    public PaxosLeadershipCreator(
+            TimeLockInstallConfiguration install,
```

Comment:
Again, if this is Paxos specific let's just take the argument as such and leave the impetus on the consumer